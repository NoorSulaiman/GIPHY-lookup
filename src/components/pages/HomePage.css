#clearSearch:hover{
    background-color: #cacbcd;
    cursor: default;
}

@media only screen and (max-width:786px){
    #clearSearch{
        margin-bottom: 2%;
    }
}