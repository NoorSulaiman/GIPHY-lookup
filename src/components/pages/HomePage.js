import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Divider, Message, Card, Button, Icon } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { searchGifs, clearSearch } from '../../actions/search';
import { addToFavorites, removeFromFavorites } from '../../actions/favorite';
import SearchGifsForm from '../forms/SearchGifsForm';
import SearchGifCard from '../templates/SearchGifCard';
import './HomePage.css';

export const HomePage = (props) => {
    const submit = query => props.searchGifs(query);
    const clearSearchResults = () => props.clearSearch();
    const toggleFavorite = (data) => {
        if (!data.isFave) {
            props.addToFavorites({ ...data, isFave: true })
        } else {
            props.removeFromFavorites(data.id)
        }
    };
    const results = props.searchResult;
    const errors = props.searchError;
    return (
        <Grid centered stackable columns={2}>
            <Grid.Row />
            <Grid.Column>
                <SearchGifsForm submit={submit} />
            </Grid.Column>
            {results && results.length > 1 && <Grid.Row centered columns={1}>
                <Divider />
                <Grid.Column >
                    <Button id="clearSearch" size='small' floated='right' onClick={clearSearchResults}>
                        <Icon name='times' />
                        Clear Search
                        </Button>
                </Grid.Column>
            </Grid.Row>}
            <Grid.Row centered>
                <Card.Group stackable itemsPerRow={4}>
                    {results.length > 0 && results.map(gif =>
                        <SearchGifCard
                            key={gif.id}
                            id={gif.id}
                            toggleFavorite={toggleFavorite}
                            title={gif.title}
                            imgUrl={gif.imgUrl}
                            isFave={gif.isFave}
                        />,
                    )}
                </Card.Group>
                {!results[0] && errors.isNoResults && <Message size='big'>No search results found!</Message>}
                {errors.length > 12 && <Message size='big' negative>{errors}</Message>}
            </Grid.Row>
        </Grid>
    );
}

HomePage.propTypes = {
    searchGifs: PropTypes.func.isRequired,
    searchResult: PropTypes.arrayOf(PropTypes.shape({
        title: PropTypes.string.isRequired,
        id: PropTypes.string.isRequired,
        imgUrl: PropTypes.string.isRequired,
        isFave: PropTypes.bool.isRequired,
    }).isRequired).isRequired,
    searchError: PropTypes.string.isRequired,
    clearSearch: PropTypes.func.isRequired,
    addToFavorites: PropTypes.func.isRequired,
    removeFromFavorites: PropTypes.func.isRequired,
};

function mapStateToProps(state) {
    return {
        searchResult: state.search.searchResult,
        searchError: state.search.searchError,
    };
};

export default connect(mapStateToProps, { searchGifs, clearSearch, addToFavorites, removeFromFavorites })(HomePage);
