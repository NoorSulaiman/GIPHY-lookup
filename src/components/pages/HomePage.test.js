import React from 'react';
import { shallow } from 'enzyme';
import { HomePage } from './HomePage';

const clickFn = jest.fn();

it('should render HomePage correctly with given props & with no SearchGifCard', () => {
    const component = shallow(<HomePage
        searchGifs={clickFn}
        searchResult={[]}
        searchError={{}}
        clearSearch={clickFn}
        addToFavorites={clickFn}
        removeFromFavorites={clickFn}
        pagination={0} />);
    expect(component).toMatchSnapshot();
    expect(component.find('SearchGifsForm').length).toBe(1);
    expect(component.find('SearchGifCard').length).toBe(0);
    expect(component.find('Message').length).toBe(0);
});

