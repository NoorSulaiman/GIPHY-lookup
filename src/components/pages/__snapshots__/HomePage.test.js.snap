// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomePage correctly with given props & with no SearchGifCard 1`] = `
<Grid
  centered={true}
  columns={2}
  stackable={true}
>
  <GridRow />
  <GridColumn>
    <SearchGifsForm
      submit={[Function]}
    />
  </GridColumn>
  <GridRow
    centered={true}
  >
    <CardGroup
      itemsPerRow={4}
      stackable={true}
    />
  </GridRow>
</Grid>
`;
